app/controllers/works_controller.rb

```diff
@@ -361,107 +347,79 @@ def edit_tags
 
   # PUT /works/1
   def update
-    set_edit_form_fields
+    if params[:cancel_button]
+      return cancel_posting_and_redirect
+    end
 
+    @work.preview_mode = !!(params[:preview_button] || params[:edit_button] ||
```

Comment:
The double `!!` is throwing me off here -- could you translate this from code to English for me?

Reply:
It's basically a type-conversion to ensure that `@work.preview_mode` is true or false instead of `"Preview"`, `"Edit"`, `"Cancel"`, or `nil`. A single `!` acts as negation and forces the results to be either true or false, so a double-negation cancels out the negation and just converts to a boolean. I'm pretty sure it would work without the double negation -- I only added it because at one point I tried to print out `@work.preview_mode` and found the fact that it held a string confusing.